### PR TITLE
Add interactor

### PR DESCRIPTION
**Pull request by jinSasaki**
Add interactors with tests

- [x] EvaluationInteractor
- [x] EventInteractor

### CODE REVIEW

File: Bucketeer2Tests/EvaluationInteractorTests.swift [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+                )
+                XCTAssertEqual(evaluation, .mock1)
+
+                expectation.fulfill()
+            }
+        }
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testRefrechCache() throws {
```
Comment (cre8ivejp) [inline Bucketeer2Tests/EvaluationInteractorTests.swift:191]: nit: testRefrechCache -> testRefreshCache

File: Bucketeer2Tests/EventInteractorTests.swift [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+        let idGenerator = MockIdGenerator(identifier: "id")
+        let addedEvents: [Event] = [.mockEvaluation1, .mockGoal1]
+        let dao = MockEventDao()
+        try dao.add(events: addedEvents)
+        let api = MockApiClient(registerEventsHandler: { events, completion in
+            XCTAssertEqual(events.count, 2)
+            XCTAssertEqual(events, addedEvents)
+            completion?(.success(.init(data: .init(
+                errors: [
+                    Event.mockEvaluation1.id: .init(retriable: true, message: "message")
```
Comment (cre8ivejp) [inline Bucketeer2Tests/EventInteractorTests.swift:492]: Could you also verify that Event.mockEvaluation1 is still in the DB?

https://github.com/bucketeer-io/android-client-sdk/blob/main/bucketeer/src/test/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractorTest.kt#L617-L648

Comment (jinSasaki) [reply Bucketeer2Tests/EventInteractorTests.swift:496]: Does this follow your point?

- Verify 2 events before `sendEvents` https://github.com/ca-dp/bucketeer-ios-sdk/pull/85/files#diff-b1760fb1974c08fa44a0e379b736e22d188e107eaa07512c4e4470db068e3de8R488-R489
- Verify that 1 event remains after `sendEvents` https://github.com/ca-dp/bucketeer-ios-sdk/pull/85/files#diff-b1760fb1974c08fa44a0e379b736e22d188e107eaa07512c4e4470db068e3de8R511-R512

File: Bucketeer2/Sources/Internal/Event/EventInteractor.swift [hunk 3]
```diff
@@ -115,7 +115,6 @@ final class EventInteractor {
     }
 
     func trackFetchEvaluationsFailure(featureTag: String, error: BKTError) throws {
-
```
Comment (jinSasaki) [inline Bucketeer2/Sources/Internal/Event/EventInteractor.swift]: removed unnecessary new line

File: Bucketeer2Tests/EvaluationInteractorTests.swift [hunk 4]
```diff
@@ -188,7 +188,7 @@ final class EvaluationInteractorTests: XCTestCase {
         wait(for: [expectation], timeout: 1)
     }
 
-    func testRefrechCache() throws {
+    func testRefreshCache() throws {
```
Comment (jinSasaki) [inline Bucketeer2Tests/EvaluationInteractorTests.swift:191]: fix typo

File: Bucketeer2Tests/EventInteractorTests.swift (hunk 2 above)
Comment (cre8ivejp) [reply Bucketeer2Tests/EventInteractorTests.swift:496]: Yes, it's correct!
Thank you!

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**